bound/kt/src/test/kotlin/web5/sdk/dids/DidTest.kt

```diff
@@ -1,22 +1,249 @@
 package web5.sdk.dids
 
-import org.junit.jupiter.api.Assertions.assertEquals
-import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.*
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.fail
+import web5.sdk.UnitTestSuite
+import web5.sdk.rust.Web5Exception
 
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 class DidTest {
 
+  private val testSuite = UnitTestSuite("did_new")
+
+  @AfterAll
+  fun verifyAllTestsIncluded() {
+    if (testSuite.tests.isNotEmpty()) {
+      println("The following tests were not included or executed:")
+      testSuite.tests.forEach { println(it) }
+      fail("Not all tests were executed! ${this.testSuite.tests}")
+    }
+  }
+
+  @Test
+  fun test_did_empty_string_should_error() {
+    this.testSuite.include()
+    val uri = ""
+
+    val exception = assertThrows<Web5Exception.Exception> {
+      Did.parse(uri)
+    }
+
+    assertEquals("parameter error identifier regex match failure $uri", exception.msg)
+  }
+
+  @Test
+  fun test_did_incomplete_scheme_should_error() {
+    this.testSuite.include()
+    val uri = "did:"
+
+    val exception = assertThrows<Web5Exception.Exception> {
+      Did.parse(uri)
+    }
+
+    assertEquals("parameter error identifier regex match failure $uri", exception.msg)
+  }
+
+  @Test
+  fun test_did_missing_id_part_should_error() {
+    this.testSuite.include()
+    val uri = "did:uport"
```

Comment:
works, but kind of strange to be using uport

Reply:
most of this is copy pasta from web5-go, so idunno what the origin of that is but meh it's fine right?